satellite/satellitedb/overlaycache.go

```diff
@@ -137,6 +205,77 @@ func (cache *overlaycache) queryFilteredNodes(ctx context.Context, excluded []st
 	return nodes, rows.Err()
 }
 
+func (cache *overlaycache) postgresQueryNodes(ctx context.Context, exclude overlay.Exclude, count int, safeQuery string, distinctIP bool, args ...interface{}) (_ []*pb.Node, err error) {
+	if count == 0 {
+		return nil, nil
+	}
+
+	safeExcludeNodes := ""
+	if len(exclude.Nodes) > 0 {
+		safeExcludeNodes = ` AND id NOT IN (?` + strings.Repeat(", ?", len(exclude.Nodes)-1) + `)`
+		for _, id := range exclude.Nodes {
+			args = append(args, id.Bytes())
+		}
+	}
+
+	safeExcludeIPs := ""
+	if len(exclude.IPs) > 0 {
+		safeExcludeIPs = ` AND last_ip NOT IN (?` + strings.Repeat(", ?", len(exclude.IPs)-1) + `)`
```

Comment:
Did you also try `last_ip <> any(values(?,?,?,?,?,?`?

Reply:
I did. It appeared to be on average a bit slower, however that could be due to the table I'm running the query against. I have the query plans if you'd like to compare them